### DISCUSSION TITLE
pdfjs and error Error rendering PDF: TypeError: Cannot read from private field

### DISCUSSION THREAD

**Discussion by cylace** [mozilla/pdf.js#18340] [Q&A]
Hi, i will show preview pdf with pdf.js (project wit laravel backend and vue3js frontend). I've that function: `const renderPdf = async (url) => {
    try {
        console.log('Fetching document from URL:', url);
        const loadingTask = pdfjsLib.getDocument(url);
        pdfDoc.value = await loadingTask.promise;
        console.log('Document loaded successfully', pdfDoc.value);

        if (pdfDoc.value.numPages === 0) {
            throw new Error('PDF has no pages');
        }

        const pageNumber=2;
        const page = await pdfDoc.value.getPage(pageNumber);
        console.log('Page loaded successfully', page);

        const viewport = page.getViewport({ scale: 1.5 });
        const canvas = pdfCanvas.value;
        const context = canvas.getContext('2d');

        canvas.width = viewport.width;
        canvas.height = viewport.height;

        const renderContext = {
            canvasContext: context,
            viewport,
        };

        console.log('Rendering page...');
        const renderTask = page.render(renderContext);
        await renderTask.promise;
        console.log('Page rendered successfully');
    } catch (error) {
        console.error('Error rendering PDF:', error);
    }
};
and on getPage i've always same error. I don't know why but i can see my pdf on another page. My objectif is to have my pdf with laravel dompdf and for preview on my page web with pdf js but i've that error: 
`
DetailBusinessCard.vue:174 Error rendering PDF: TypeError: Cannot read from private field
    at __accessCheck (chunk-TIUEEL27.js?v=d20829a4:41:11)
    at __privateGet (chunk-TIUEEL27.js?v=d20829a4:44:3)
    at Proxy.getPage (api.js:2923:23)
    at Proxy.getPage (api.js:833:28)
    at renderPdf (BusinessCard.vue:154:41)
    at async generatePreviewPdf (BusinessCard.vue:131:13)`

 I needed help please, thank you so much

**Comment by qwecxzpoi** [comment]
if you use `vue 3`, you can use `const pdfDoc = shallowRef<PDFDocumentProxy>()` or `toRaw(pdfDoc).getPage(pageNumber)`
